core/CronArchive/SegmentArchiving.php

```diff
@@ -111,16 +111,25 @@ public function getReArchiveSegmentStartDate($segmentInfo)
          */
         list($segmentCreatedTime, $segmentLastEditedTime) = $this->getCreatedTimeOfSegment($segmentInfo);
 
-        if ($this->processNewSegmentsFrom == SegmentArchiving::CREATION_TIME && !empty($segmentCreatedTime)) {
+        if ($this->processNewSegmentsFrom == SegmentArchiving::CREATION_TIME) {
+            if (empty($segmentCreatedTime)) {
+                return null;
+            }
             $this->logger->debug("process_new_segments_from set to segment_creation_time, oldest date to process is {time}", array('time' => $segmentCreatedTime));
 
             return $segmentCreatedTime;
-        } else if ($this->processNewSegmentsFrom == SegmentArchiving::LAST_EDIT_TIME && !empty($segmentLastEditedTime)) {
+        } else if ($this->processNewSegmentsFrom == SegmentArchiving::LAST_EDIT_TIME) {
+            if (empty($segmentLastEditedTime)) {
+                return null;
+            }
             $this->logger->debug("process_new_segments_from set to segment_last_edit_time, segment last edit time is {time}",
                 array('time' => $segmentLastEditedTime));
 
             return $segmentLastEditedTime;
-        } else if (preg_match("/^editLast([0-9]+)$/", $this->processNewSegmentsFrom, $matches) && !empty($segmentLastEditedTime)) {
+        } else if (preg_match("/^editLast([0-9]+)$/", $this->processNewSegmentsFrom, $matches)) {
+            if (empty($segmentLastEditedTime)) {
+                return null;
```

Comment:
was going to say here we could maybe fall back to `Date::now()` but I suppose it's not really needed anyway as we should always have an edited time.

Reply:
An empty edit time shouldn't happen, indeed. Nevertheless assuming `Date::now()` when it wouldn't be set for any reason would only be correct if we assume this method is only called when the segment had been edited/created.

Reply:
👍  all good, it's not needed. LGTM